Packs/GitHubITAdmin/Integrations/GitHub_IT_Admin/GitHub_IT_Admin.py

```diff
@@ -0,0 +1,615 @@
+
+        github_user = scim
+
+        custom_mapping = None
+        if args.get('customMapping'):
+            custom_mapping = json.loads(args.get('customMapping'))
+        elif CUSTOM_MAPPING_CREATE:
+            custom_mapping = json.loads(CUSTOM_MAPPING_CREATE)
+
+        if custom_mapping and extension_schema:
+            for key, value in custom_mapping.items():
+                # key is the attribute name in input scim. value is the attribute name of app profile
+                user_extension_data = extension_schema.get(key)
+                if user_extension_data:
+                    github_user[value] = user_extension_data
+
+        return github_user
+
+    # Builds a new user github profile dict with pre-defined keys and custom mapping (for user)
+    def build_github_update_user(self, args, github_user, scim):
+        extension_schema = scim.get(SCIM_EXTENSION_SCHEMA)
+        custom_mapping = None
+        if args.get('customMapping'):
+            custom_mapping = json.loads(args.get('customMapping'))
+        elif CUSTOM_MAPPING_UPDATE:
+            custom_mapping = json.loads(CUSTOM_MAPPING_UPDATE)
+
+        if custom_mapping and extension_schema:
+            for key, value in custom_mapping.items():
+                # key is the attribute name in input scim. value is the attribute name of app profile
+                user_extension_data = extension_schema.get(key)
+                if user_extension_data:
+                    github_user[value] = user_extension_data
+
+        return github_user
+
+
+class OutputContext:
+    """
+        Class to build a generic output and context.
+    """
+
+    def __init__(self, success=None, active=None, iden=None, username=None, email=None, errorCode=None,
+                 errorMessage=None, details=None):
+        self.instanceName = demisto.callingContext['context']['IntegrationInstance']
+        self.brand = demisto.callingContext['context']['IntegrationBrand']
+        self.command = demisto.command().replace('-', '_').title().replace('_', '')
+        self.success = success
+        self.active = active
+        self.iden = iden
+        self.username = username
+        self.email = email
+        self.errorCode = errorCode
+        self.errorMessage = errorMessage
+        self.details = details
+        self.data = {
+            "brand": self.brand,
+            "instanceName": self.instanceName,
+            "success": success,
+            "active": active,
+            "id": iden,
+            "username": username,
+            "email": email,
+            "errorCode": errorCode,
+            "errorMessage": errorMessage,
+            "details": details
+        }
+
+
+def verify_and_load_scim_data(scim):
+
+    try:
+        scim = json.loads(scim)
+    except Exception:
+        pass
+    if type(scim) != dict:
+        raise Exception("SCIM data is not a valid JSON ")
+    return scim
+
+
+def map_scim(scim):
+    try:
+        scim = json.loads(scim)
```

Comment:
Isn't this a duplicate of the `verify_and_load_scim_data` method? I see map_scim is always called after the verification, and you are already sending the dict - am I correct?

Reply:
We have removed the duplicate lines of code.